nearcache_map.go

```diff
@@ -182,6 +229,44 @@ func (ncm *nearCacheMap) Get(ctx context.Context, m *Map, key interface{}) (inte
 	return value, nil
 }
 
+func (ncm *nearCacheMap) GetAll(ctx context.Context, m *Map, keys []interface{}) ([]types.Entry, error) {
+	// see: com.hazelcast.client.map.impl.nearcache.NearCachedClientMapProxy#getAllInternal
+	entries := make([]types.Entry, len(keys))
+	missKeys, err := ncm.populateResultFromNearCache(keys, entries)
+	if err != nil {
+		return nil, fmt.Errorf("nearCacheMap.GetAll: populating result from near cache: %w", err)
+	}
+	if len(missKeys) == 0 {
+		return entries, nil
+	}
+	keyDatas := make([]serialization.Data, len(keys))
+	for i, k := range missKeys {
+		kd, err := ncm.ss.ToData(k)
+		if err != nil {
+			return nil, err
+		}
+		keyDatas[i] = kd
+	}
+	resMap, err := ncm.getNearCacheReservations(keys, keyDatas)
+	if err != nil {
+		return nil, err
+	}
+	partitionToKeys, err := m.partitionToKeys(keys, false)
+	if err != nil {
+		return nil, err
+	}
+	defer ncm.releaseRemainingReservedKeys(resMap)
+	pairs, err := m.getAllFromRemote(ctx, len(missKeys), partitionToKeys)
+	if err != nil {
+		return nil, fmt.Errorf("nearCacheMap.GetAll: getting keys from remote: %w", err)
+	}
+	keyCount, err := ncm.populateResultFromRemote(pairs, entries, resMap)
+	if err != nil {
+		return nil, err
+	}
+	return entries[:keyCount], nil
```

Comment:
I don't understand the reason for slicing the `entries`? Can you please explain?